exercises/hangman/.meta/src/reference/java/Hangman.java

```diff
@@ -0,0 +1,89 @@
+import io.reactivex.Observable;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.joining;
+
+class Hangman {
+
+    Observable<Output> play(
+            Observable<String> words,
+            Observable<String> letters) {
+        return Observable
+            .combineLatest(
+                words,
+                letters.startWith(""),
+                (word, letter) -> new AbstractMap.SimpleEntry<>(word, letter))
+            .scan(
+                new Output(
+                    null,
+                    null,
+                    Collections.emptySet(),
+                    Collections.emptySet(),
+                    Collections.emptyList(),
+                    null),
```

Comment:
Constructors that require you to pass in null for some of the parameters is pretty good sign that you should be using either (1) constructor overloads or (2) static factory methods.

Constructor overloads:
```java
Output() {
  // can pass to another method or set some fields directly
  this(...);
}
```

Static factory method:

```java
static Output empty() {
  return new Output(...empty fields...);
}
```

Reply:
True. I'll go the static factory and see if more are relevant.